logic/src/commonMain/kotlin/com/wire/kalium/logic/data/team/TeamRepository.kt

```diff
@@ -80,9 +81,35 @@ internal class TeamDataSource(
         teamsApi.getTeamInfo(teamId = teamId.value)
     }.map { teamDTO ->
         teamMapper.fromDtoToEntity(teamDTO)
-    }.flatMap { teamEntity ->
-        wrapStorageRequest { teamDAO.insertTeam(team = teamEntity) }.map {
-            teamMapper.fromDaoModelToTeam(teamEntity)
+    }.map { teamEntity ->
+        teamDAO.insertTeam(team = teamEntity)
+
+        teamMapper.fromDaoModelToTeam(teamEntity)
+    }
+
+    override suspend fun fetchMembersByTeamId(teamId: TeamId, userDomain: String): Either<CoreFailure, Unit> = wrapApiRequest {
+        teamsApi.getTeamMembers(
+            teamId = teamId.value,
+            limitTo = null
+        )
+    }.map { teamMemberList ->
+        /**
+         * If hasMore is true, then this result should be discarded and not stored locally,
+         * otherwise the user will see random team members when opening the search UI.
+         * If the result has has_more field set to false, then these users are stored locally to be used in a search later.
+         */
+        if (teamMemberList.hasMore.not()) {
+            teamMemberList.members.map { teamMember ->
+                val userId = QualifiedIDEntity(teamMember.nonQualifiedUserId, userDomain)
+                val userType = userTypeEntityTypeMapper.teamRoleCodeToUserType(teamMember.permissions?.own)
+                userId to userType
+            }
+        } else {
+            listOf()
+        }
```

Comment:
The point of this change is exactly to make sure that we fetch everyone eagerly. InfiniteMonkeys, for example, could have teams of 10k users.

Fetching only a part of it could lead to silent failures or unexpected behaviour.
When the flag is `true`, we should consume all the pages to make sure we have all users.